Formula/glibc.rb

```diff
@@ -95,41 +95,188 @@ class Glibc < Formula
-  fails_with gcc: "4.6"
+  MINIMUM_MAKE_VERSION = "4.0".freeze
+  MINIMUM_GCC_VERSION = "6.2".freeze
+  INSTALLED_MAKE_VERSION = Utils.safe_popen_read("make", "--version").lines.first.split.third.freeze
+  INSTALLED_GCC_VERSION = Utils.safe_popen_read("gcc", "-dumpversion").freeze
 
-  def install
-    # Fix Error: `loc1@GLIBC_2.2.5' can't be versioned to common symbol 'loc1'
-    # See https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=869717
-    # Fixed in glibc 2.24
-    inreplace "misc/regexp.c", /^(char \*loc[12s]);$/, "\\1 __attribute__ ((nocommon));"
+  if Version.new(INSTALLED_MAKE_VERSION) < MINIMUM_MAKE_VERSION
+    resource "make" do
+      url "https://ftp.gnu.org/gnu/make/make-4.3.tar.gz"
+      sha256 "e05fdde47c5f7ca45cb697e973894ff4f5d79e13b750ed57d7b66d8defc78e19"
+    end
+  end
+
+  if which("python3").blank?
+    resource "python3" do
+      url "https://www.python.org/ftp/python/3.10.5/Python-3.10.5.tgz"
+      sha256 "18f57182a2de3b0be76dfc39fdcfd28156bb6dd23e5f08696f7492e9e3d0bf2d"
+    end
+  end
+
+  if which("bison").blank?
+    if which("m4").blank?
+      resource "m4" do
+        url "https://ftp.gnu.org/gnu/m4/m4-1.4.19.tar.xz"
+        sha256 "63aede5c6d33b6d9b13511cd0be2cac046f2e70fd0a07aa9573a04a82783af96"
+      end
+    end
+
+    resource "bison" do
+      url "https://ftp.gnu.org/gnu/bison/bison-3.8.2.tar.xz"
+      sha256 "9bba0214ccf7f1079c5d59210045227bcf619519840ebfa80cd3849cff5a5bf2"
+    end
+  end
+
+  if Version.new(INSTALLED_GCC_VERSION) < MINIMUM_GCC_VERSION
+    unless File.exist?("/usr/bin/bzip2")
+      resource "bzip2" do
+        url "https://sourceware.org/pub/bzip2/bzip2-1.0.8.tar.gz"
+        sha256 "ab5a03176ee106d3f0fa90e381da478ddae405918153cca248e682cd0c4a2269"
+      end
+    end
+
+    resource "gcc" do
+      url "https://ftp.gnu.org/gnu/gcc/gcc-10.4.0/gcc-10.4.0.tar.xz"
+      sha256 "c9297d5bcd7cb43f3dfc2fed5389e948c9312fd962ef6a4ce455cff963ebe4f1"
+    end
+
+    resource "mpfr" do
+      url "ftp://gcc.gnu.org/pub/gcc/infrastructure/mpfr-3.1.6.tar.bz2"
+      sha256 "cf4f4b2d80abb79e820e78c8077b6725bbbb4e8f41896783c899087be0e94068"
+    end
+
+    resource "mpc" do
+      url "ftp://gcc.gnu.org/pub/gcc/infrastructure/mpc-1.0.3.tar.gz"
+      sha256 "617decc6ea09889fb08ede330917a00b16809b8db88c29c31bfbb49cbf88ecc3"
+    end
+
+    resource "gmp" do
+      url "ftp://gcc.gnu.org/pub/gcc/infrastructure/gmp-6.1.0.tar.bz2"
+      sha256 "498449a994efeba527885c10405993427995d3f86b8768d8cdf8d9dd7c6b73e8"
+    end
+  end
 
+  def install
     # Setting RPATH breaks glibc.
     %w[
       LDFLAGS LD_LIBRARY_PATH LD_RUN_PATH LIBRARY_PATH
       HOMEBREW_DYNAMIC_LINKER HOMEBREW_LIBRARY_PATHS HOMEBREW_RPATH_PATHS
     ].each { |x| ENV.delete x }
 
+    ENV["CC"] = DevelopmentTools.locate("gcc")
+    ENV["CXX"] = DevelopmentTools.locate("g++")
+    ENV.prepend_path "PATH", buildpath/"bootstrap/bin"
+
+    if Version.new(INSTALLED_MAKE_VERSION) < MINIMUM_MAKE_VERSION
```

Comment:
This check is repeated multiple times, would be nice to make a variable or method.

Reply:
Actually if I put this in a Requirement then maybe that will be cleaner?  I’m still figuring out how to use Requirement so that may be a nonsensical idea.

Reply:
> I’m not sure how to define the method in such a way that it can be used to decide if the resource needs to be downloaded.

Perhaps needs to be a class method?



> Actually if I put this in a Requirement then maybe that will be cleaner?

Yeh, may be 👍🏻